imap_data_access/file_validation.py

```diff
@@ -68,6 +69,51 @@ def __init__(self, filename: str | Path, data_dir: Path | None = None):
         if self.error_message:
             raise self.InvalidScienceFileError(f"{self.error_message}")
 
+    @classmethod
+    def generate_from_inputs(
+        cls,
+        instrument: str,
+        level: str,
```

Comment:
```suggestion
        data_level: str,
```
To be consistent with how we are naming this elsewhere